scenariolib/event_pageview.go

```diff
@@ -63,17 +70,22 @@ func newViewEvent(e *JSONEvent, c *Config) (*ViewEvent, error) {
 // Execute the view event, sending a view event to the usage analytics
 func (ve *ViewEvent) Execute(v *Visit) error {
 	if v.LastResponse == nil {
-		return errors.New("LastResponse was nil, cannot send a pageview. Please use a search event before.")
+		return errors.New("No query before pageView event, use a search event first")
 	}
 	if v.LastResponse.TotalCount < 1 {
-		Warning.Printf("Last query %s returned no results cannot click", v.LastQuery.Q)
+		Warning.Printf("Last query %s returned no results cannot send view event", v.LastQuery.Q)
 		return nil
 	}
 
 	if rand.Float64() <= ve.probability { // test if the event will exectute according to probability
 		ve.computeClickRank(v)
 
-		err := v.sendViewEvent(ve.clickRank, ve.contentType, ve.pageViewField)
+		if ve.clickRank > v.LastResponse.TotalCount {
+			Warning.Printf("PageView index out of bounds, not sending event")
```

Comment:
Would clicking on the last result of the page be a better solution than _not_ sending the event at all?

Can a quick `Math.min` (or whatever) on `ve.clickRank` and `v.LastResponse.TotalCount` give a better simulation...?

Reply:
It's already done in the `computeClickRank()` function. This is an actual error that should never happen :(